staging/src/k8s.io/apimachinery/pkg/apis/meta/v1/unstructured/unstructured.go

```diff
@@ -200,22 +201,28 @@ func getNestedField(obj map[string]interface{}, fields ...string) interface{} {
 	return val
 }
 
-func getNestedString(obj map[string]interface{}, fields ...string) string {
-	if str, ok := getNestedField(obj, fields...).(string); ok {
+// NestedString returns the string value of a nested field.
+// Returns empty string if value is not found or is not a string.
+func NestedString(obj map[string]interface{}, fields ...string) string {
+	if str, ok := NestedField(obj, fields...).(string); ok {
```

Comment:
I would prefer to see `error`s in the result if the type is wrong.

Same for the other type-casts below.